*: remove unused pre-23.1 gates

RaduBerinde opened the pull request:
Fixes #112501.  Note that gates inside `pkg/sql` were already removed in #114307.

#### kv: remove pre-23.1 gates

Epic: none
Release note: None

#### server: remove pre-23.1 gates

Epic: none
Release note: None

#### ccl: remove pre-23.1 gates

Epic: none
Release note: None

#### jobs: remove pre-23.1 gates

Note that there is some follow-up cleanup remaining, and it is not as
simple as the TODO comments suggest. The remaining work is tracked
in #112545.

Epic: none
Release note: None

#### *: remove last remaining pre-23.1 gates

Epic: none
Release note: None

#### clusterversion: remove unused pre-23.1 gates

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114468)
<!-- Reviewable:end -->

RaduBerinde committed:
jobs: remove pre-23.1 gates

Note that there is some follow-up cleanup remaining, and it is not as
simple as the TODO comments suggest. The remaining work is tracked
in #112545.

Epic: none
Release note: None

RaduBerinde committed:
ccl: remove pre-23.1 gates

Epic: none
Release note: None

RaduBerinde committed:
clusterversion: remove unused pre-23.1 gates

Epic: none
Release note: None

RaduBerinde committed:
server: remove pre-23.1 gates

Epic: none
Release note: None

RaduBerinde committed:
*: remove last remaining pre-23.1 gates

Epic: none
Release note: None

RaduBerinde committed:
kv: remove pre-23.1 gates

Epic: none
Release note: None

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, 6 of 6 files at r2, 6 of 6 files at r3, 5 of 5 files at r4, 5 of 5 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114468#-:-NjJLO0J2Bsy_l5MiSlk:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @abarganier, @adityamaru, @andyyang890, @DarrylWong, @jayshrivastava, @RaduBerinde, and @renatolabs)

___
*[`pkg/ccl/backupccl/backup_planning_tenant.go` line 23 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/114468#-NjJJCaL5h1T9t8-repF:-NjJJCaL5h1T9t8-repG:b-nm8lrr) ([raw file](https://github.com/cockroachdb/cockroach/blob/ae047549aeb9da512b22c9d180a92762fa39b590/pkg/ccl/backupccl/backup_planning_tenant.go#L23)):*
> ```Go
> )
> 
> func tenantMetadataQuery(ctx context.Context, settings *cluster.Settings) string {
> ```

nit: could make this a constant string.
___
*[`pkg/jobs/adopt.go` line 269 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114468#-NjJJxZm0NPLuTS_-z0J:-NjJJxZn-4eYkkAGVesr:b-e75roa) ([raw file](https://github.com/cockroachdb/cockroach/blob/ed163e60d8c9527a1ec637772549ae8cdea7efaf/pkg/jobs/adopt.go#L269)):*
> ```Go
> 		resumeQuery = resumeQueryWithBackoff
> 	} else {
> 		resumeQuery = deprecatedResumeQueryWithBackoff
> ```

nit: we should remove `deprecatedResumeQueryBaseCols` and `deprecatedResumeQueryWithBackoff` with the TODO on top of them.
___
*[`pkg/jobs/update.go` line 393 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114468#-NjJKyf9AOmCjDfpX6qy:-NjJKyf9AOmCjDfpX6qz:b-2x0bnf) ([raw file](https://github.com/cockroachdb/cockroach/blob/5313f6b9542c1191094b70e2cf34b434acb6f878/pkg/jobs/update.go#L393)):*
> ```Go
> 		fromForUpdate = from + `FOR UPDATE`
> 
> 		// TODO(adityamaru): Delete deprecated queries once we are outside the 22.2
> ```

nit: should address this TODO.


<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114468#-:-NjJOGDo3QvuFiXWnEZH:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @abarganier, @adityamaru, @andyyang890, @DarrylWong, @jayshrivastava, @renatolabs, and @yuzefovich)

___
*[`pkg/jobs/update.go` line 393 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/114468#-NjJKyf9AOmCjDfpX6qy:-NjJQ7MpC75UkJKWh-Y2:blq7vew) ([raw file](https://github.com/cockroachdb/cockroach/blob/5313f6b9542c1191094b70e2cf34b434acb6f878/pkg/jobs/update.go#L393)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: should address this TODO.
</blockquote></details>

Done. I also improved the code below a bit.


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
